src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx

```diff
@@ -0,0 +1,211 @@
+'use client';
+import {
+  Accordion,
+  AccordionItem,
+  AccordionContent,
+  AccordionTrigger,
+} from '@/components/ui/accordion';
+import { Checkbox } from '@/components/ui/checkbox';
+import { Info } from 'lucide-react';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { TaskDescriptionDrawer } from '../ui/drawers/TaskDescriptionDrawer';
+import { Button } from '../ui/button';
+import { LoadingSpinner } from '../ui/loadingSpinner';
+import { toast } from '../ui/use-toast';
+import { cn } from '@/lib/utils';
+
+interface AccordionAssignmentSheetProps {
+  rotationData: {
+    rotations: {
+      [key: number]: {
+        startDate: string;
+        endDate: string;
+        categories: {
+          id: string;
+          name: string;
+          tasks: {
+            id: string;
+            title: string;
+            description?: string;
+            isCompleted?: boolean;
+          }[];
+        }[];
+      };
+    };
+  };
+}
+
+export const AccordionAssignmentSheet = ({
+  rotationData,
+}: AccordionAssignmentSheetProps) => {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const [isChecked, setIsChecked] = useState<{ [taskId: string]: boolean }>({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentTask, setCurrentTask] = useState<{
+    title: string;
+    description: string;
+  }>({ title: '', description: '' });
+
+  const handleCheckboxChange = (taskId: string) => {
+    setIsChecked((prevChecked) => ({
+      ...prevChecked,
+      [taskId]: !prevChecked[taskId],
+    }));
+  };
+
+  const handleAllCheckedChange = (categoryId: string) => {
+    // Get all task IDs for the category
+    const taskIds = Object.values(rotationData.rotations).flatMap(
+      (rotation) =>
+        rotation.categories
+          .find((category) => category.id === categoryId)
+          ?.tasks.map((task) => task.id) || [],
+    );
+
+    // Check if all tasks in the category are checked
+    const allChecked = taskIds.every((taskId) => isChecked[taskId]);
+
+    // Create a new checked state with toggled values
+    const newChecked = { ...isChecked };
+    taskIds.forEach((taskId) => {
+      newChecked[taskId] = !allChecked;
+    });
+
+    // Update the checked state
+    setIsChecked(newChecked);
+  };
+
+  const { handleSubmit } = useForm();
+
+  const handleSaveClick = () => {
+    const updatedTasks = Object.values(rotationData.rotations)
+      .flatMap((rotation) =>
+        rotation.categories.flatMap((category) => category.tasks),
+      )
+      .filter((task) => isChecked[task.id])
+      .map((task) => ({
+        ...task,
+        isCompleted: isChecked[task.id] || false,
+      }));
+    console.log('updated', updatedTasks);
+  };
+
+  // TODO: Implement the onSubmit click functionality
+  const onSubmit = async () => {
```

Comment:
IMO, I think we could integrate onSubmit and onSaveClick since they trigger at the same timing, which would simplify things.